src/legacy_wrappers/mongo_client.js

```diff
@@ -7,6 +7,18 @@ Object.defineProperty(module.exports, '__esModule', { value: true });
 
 module.exports.makeLegacyMongoClient = function (baseClass) {
   class LegacyMongoClient extends baseClass {
+    // constructor adds client metadata before constructing final client
+    constructor(connectionString, options) {
+      options = { ...options };
+      if (options.driverInfo != null && typeof options.driverInfo.name === 'string') {
+        options.driverInfo.name += '|mongodb-legacy';
```

Comment:
I think that prepending would be correct here, so that you would end up with `mongodb-driver|mongodb-legacy|mongoose` rather than `mongodb-driver|mongoose|mongodb-legacy`.

Reply:
good point, fixed, now prepending